zerver/lib/generateData.py

```diff
@@ -0,0 +1,273 @@
+from __future__ import absolute_import
+from __future__ import print_function
+import itertools
+import json
+import random
+from typing import List, Dict, Any, Text, Optional
+from six.moves import range
+
+def loadConfig():
+    # type: () -> Dict [str, Any]
+    infile = open("zerver/lib/config.generate_data.json", "r")
+    config = json.loads(infile.read())
+
+    return config
+
+
+def getStreamTitle(gens):
+    # type: (Dict[str, Any]) -> str
+
+    # the secret to generating unique data is:
+    # make sure that count of each list is a different prime number
+    # with these four, we will get 15015 unique values before a dupe
+    # 15 * 11 * 7 * 5 * 3 = 15015
+
+    return next(gens["adjectives"]) + " " + next(gens["nouns"]) + " " + \
+        next(gens["connectors"]) + " " + next(gens["verbs"]) + " " + \
+        next(gens["adverbs"])
+
+
+def loadGenerators(config):
+    # type: (Dict[str, Any]) -> Dict[str, Any]
+
+    results = {}
+    cfg = config["gen_fodder"]
+
+    results["nouns"] = itertools.cycle(cfg["nouns"])             # 13 items
+    results["adjectives"] = itertools.cycle(cfg["adjectives"])   # 11 items
+    results["connectors"] = itertools.cycle(cfg["connectors"])   # 7 items
+    results["verbs"] = itertools.cycle(cfg["verbs"])             # 5 items
+    results["adverbs"] = itertools.cycle(cfg["adverbs"])         # 3 items
+    results["emojis"] = itertools.cycle(cfg["emoji"])
+    results["links"] = itertools.cycle(cfg["links"])
+
+    results["maths"] = itertools.cycle(cfg["maths"])
+    results["inline-code"] = itertools.cycle(cfg["inline-code"])
+    results["code-blocks"] = itertools.cycle(cfg["code-blocks"])
+    results["quote-blocks"] = itertools.cycle(cfg["quote-blocks"])
+
+    results["lists"] = itertools.cycle(cfg["lists"])
+
+    return results
+
+
+def checkForDupes(gens):
+    # type: (Dict[str, Any]) -> None
+
+    results = []  # type: List[str]
+
+    for i in range(106):
+        key = next(gens["verbs"]) + " " + next(gens["connectors"]) + \
+            " " + next(gens["adverbs"])
+        if key not in results:
+            results.append(key)
+            print(key)
+        else:
+            print("*******\nDupe found at {}\n{}".format(str(i), key))
+            break
```

Comment:
This function seems to not be used; maybe it was a debugging tool we don't need anymore?